datafusion/sql/src/unparser/expr.rs

```diff
@@ -169,7 +199,7 @@ impl Unparser<'_> {
     pub(super) fn new_ident(&self, str: String) -> ast::Ident {
         ast::Ident {
             value: str,
-            quote_style: self.dialect.identifier_quote_style(),
+            quote_style: Some(self.dialect.identifier_quote_style().unwrap_or('"')),
```

Comment:
Identifiers are now always quoted. This makes unparser work for columns with spaces or other strangeness without any other special handling required.